owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/property/service/PropertyServiceImpl.java

```diff
@@ -0,0 +1,345 @@
+import fr.openwide.core.jpa.exception.SecurityServiceException;
+import fr.openwide.core.jpa.exception.ServiceException;
+import fr.openwide.core.jpa.more.business.parameter.model.Parameter;
+import fr.openwide.core.jpa.more.business.parameter.service.IAbstractParameterService;
+import fr.openwide.core.jpa.more.business.property.dao.IImmutablePropertyDao;
+import fr.openwide.core.jpa.more.business.property.dao.IMutablePropertyDao;
+import fr.openwide.core.jpa.more.business.property.model.ImmutablePropertyId;
+import fr.openwide.core.jpa.more.business.property.model.ImmutablePropertyRegistryKey;
+import fr.openwide.core.jpa.more.business.property.model.MutablePropertyId;
+import fr.openwide.core.jpa.more.business.property.model.PropertyId;
+import fr.openwide.core.jpa.more.business.property.model.PropertyRegistryKey;
+import fr.openwide.core.jpa.more.config.spring.event.PropertyServiceInitEvent;
+import fr.openwide.core.spring.config.CoreConfigurer;
+
+/**
+ * Use this service to retrieve application properties instead of using {@link CoreConfigurer} or {@link IAbstractParameterService}.
+ * It handles both mutable and immutable properties ; immutable properties are retrieved from properties resources files
+ * ({@link IImmutablePropertyDao}) and mutable properties are related to {@link Parameter} ({@link IMutablePropertyDao}).
+ * @see {@link IPropertyRegistry} to register application properties.
+ */
+@Service("propertyService")
+public class PropertyServiceImpl implements IConfigurablePropertyService, ApplicationEventPublisherAware {
+
+	private final Map<PropertyRegistryKey<?>, Pair<? extends Converter<String, ?>, ? extends Supplier<?>>> propertyInformationMap = Maps.newHashMap();
+
+	@Autowired
+	private IMutablePropertyDao mutablePropertyDao;
+	
+	@Autowired
+	private IImmutablePropertyDao immutablePropertyDao;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private TransactionTemplate readOnlyTransactionTemplate;
+
+	private TransactionTemplate writeTransactionTemplate;
+
+	@PostConstruct
+	public void init() {
+		applicationEventPublisher.publishEvent(new PropertyServiceInitEvent(this));
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	@Override
+	public <T> void register(PropertyRegistryKey<T> propertyId, Converter<String, ? extends T> converter) {
+		register(propertyId, converter, (T) null);
+	}
+
+	@Override
+	public <T> void register(PropertyRegistryKey<T> propertyId, Converter<String, ? extends T> converter, final T defaultValue) {
+		register(propertyId, converter, new SerializableSupplier<T>() {
+			private static final long serialVersionUID = 1L;
+			@Override
+			public T get() {
+				return defaultValue;
+			}
+		});
+	}
+
+	@Override
+	public <T> void registerImmutable(ImmutablePropertyRegistryKey<T> propertyId, Function<String, ? extends T> function) {
```

Comment:
I think it would be better to replace: 
 * `registrer(PropertyRegistryKey, Converter)` with `registrer(MutablePropertyRegistryKey, Converter)` (limit its use to mutable properties)
 * `registerImmutable(ImmutablePropertyRegistryKey, Function)` with `register(ImmutablePropertyRegistryKey, Function)` (just rename it).

Since `Converter` implements `Function`, it should work all right, and giving the same name to these methods should not lead to any conflict when calling them.
It might requires some documentation on the interface methods, though, to inform users they can use `Converter`s when we ask for `Function`s.

What do you think?